annotations/component-annotations/src/main/java/io/dekorate/component/handler/ComponentHandler.java

```diff
@@ -150,10 +152,12 @@ private BaseConfig getKubernetesConfig() {
    * @return The component.
    */
   private Component createComponent(ComponentConfig config) {
-       return new ComponentBuilder()
+    Map<String, String> labels = resources.getLabels();
+    labels.put(Labels.APP, config.getName());
```

Comment:
LGTM but I'm just wondering if it should be done in a decorator? @iocanel

Reply:
I prefer to have as many things as possibly set via decorator, but it's not a blocker for me.

The benefit of using a decoratror is that you don't have to keep track of the origin.

Was the resource created using fallback config? It works.
Was the resource created externally? it works.

Reply:
Makes sense. Does it apply to this case though?

Reply:
For the name and labels not, as we don't do that in the rest of the modules.

Reply:
:-) ok. I agree.